backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentDatabaseHandler.java

```diff
@@ -619,6 +620,13 @@ public RequestStatus updateComponent(Component component, User user) throws SW36
             if (isComponentNameChanged) {
                 updateComponentDependentFieldsForRelease(component,referenceDocLogList,user.getEmail());
             }
+
+            if (component.getComponentType() != null && !component.getComponentType().equals(actual.getComponentType())) {
+            	if (!ComponentType.OSS.equals(component.getComponentType())) {
```

Comment:
consider adding all the condition in single if condition.

Reply:
done the changes